Date: Dec 05 23:07:54 2013

From: eb5c952a06a2a3c210d27dd494a9cd6e44a4de5a Dec 12 15:26:52 2013

Subject: [PATCH] qtwebkit: add patches for RMF

        including:
        - updated to clearRect call only.
        - Updated to do clearRec and setVideoRectangle with the proper coordinates.

Source: COMCAST


Upstream-Status: Pending

Signed-off-by: Balaji Punnuru <balaji_punnuru@cable.comcast.com>
---

Index: qtwebkit-opensource-src-5.1.1/Source/WebCore/WebCore.pri
===================================================================
--- qtwebkit-opensource-src-5.1.1.orig/Source/WebCore/WebCore.pri
+++ qtwebkit-opensource-src-5.1.1/Source/WebCore/WebCore.pri
@@ -10,10 +10,23 @@ SOURCE_DIR = $${ROOT_WEBKIT_DIR}/Source/
 QT *= network sql core-private gui-private
 
 
+DEFINES += ENABLE_DLNA_VIDEO_PLAYER
 enable?(WINDOWLESS_FLASH_PLUGIN) {
     QT *= opengl
 }
 
+#COMCAST MODIFICATION BEGIN [WEBRTC]
+enable?(MEDIA_STREAM) {
+    QT *= webrtc
+}
+
+INCLUDEPATH += \
+    $$SOURCE_DIR/Modules/mediastream \
+    $$SOURCE_DIR/platform/mediastream \
+    $$SOURCE_DIR/platform/mediastream/qt \
+
+#COMCAST MODIFICATION END [WEBRTC]
+
 WEBCORE_GENERATED_SOURCES_DIR = $${ROOT_BUILD_DIR}/Source/WebCore/$${GENERATED_SOURCES_DESTDIR}
 
 INCLUDEPATH += \
@@ -181,6 +194,12 @@ use?(GSTREAMER) {
     }
 }
 
+#COMCAST MODIFICATION BEGIN [DLNA]
+contains(DEFINES, ENABLE_DLNA_VIDEO_PLAYER) {
+        LIBS += -lrmfbase -lhnsource -lmediaplayersink -ldumpfilesink
+}
+#COMCAST MODIFICATION END [DLNA]
+
 enable?(VIDEO) {
     use?(GSTREAMER) {
         INCLUDEPATH += $$SOURCE_DIR/platform/graphics/gstreamer
